### PR TITLE
What's New in 2.0

### LINKED ISSUE #4040: What's new for 2.0
The purpose of this issue is to collect text and examples we can then put into a what's new document. 
Then once we have that, someone can open a PR with all the text/examples in this issue in an rst document.

### PR DESCRIPTION
**Pull request by dstansby**
Suggestions/comments/things that don't make sense very welcome. 

Fixes #4040

### CODE REVIEW

File: docs/whatsnew/2.0.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+sunpy uses zero-based indexing when referring to pixels, where the center of
+the bottom left pixel of a map is at ``[0, 0] * u.pix``. Several parts of the
+API have been updated to make sure this is consistently the case across the
+pacakge. In particular:
+
+- `sunpy.map.GenericMap.top_right_coord` previously had an off-by-one error
+  in the calculation of the top right coordinate. This has been fixed.
+- `sunpy.map.GenericMap.center` previously had an off-by-one error
+  in the calculation of the coordinate of the center of a map. This has been fixed.
+- `sunpy.map.GenericMap.reference_pixel` now returns a zero-based reference
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:29]: are you also going to add notes about submap?

Comment (dstansby) [reply docs/whatsnew/2.0.rst:29]: I was thinking of, but didn't since I don't think there were any pixel indexing issues in submap.

Comment (Cadair) [reply docs/whatsnew/2.0.rst:29]: Well we will probably have another section on rectangles, so we could mention it there maybe :smile:

File: docs/whatsnew/2.0.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+========
+
+The SunPy project is pleased to announce the 2.0 release of the sunpy package.
+
+
+.. _whatsnew-2.0-pixels:
+
+Fixes and clarification to pixel indexing
+=========================================
+
```
Comment (hayesla) [inline docs/whatsnew/2.0.rst:19]: we should also explicitly call out `make_fitswcs_header`

File: docs/whatsnew/2.0.rst (hunk 1 above)
Comment (dstansby) [reply docs/whatsnew/2.0.rst:29]: That sounds good 👍

File: docs/whatsnew/2.0.rst (hunk 2 above)
Comment (Cadair) [reply docs/whatsnew/2.0.rst:29]: Didn't we backport that fix? If we did it's not really a what's new in 2.0 as we already released it.

File: docs/whatsnew/2.0.rst [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+We have refined our approach for heliographic Carrington coordinates to best support high-resolution imagery of the Sun, including from observatories that are at distances from the Sun that is significantly different from 1 AU (e.g., `Solar Orbiter <https://en.wikipedia.org/wiki/Solar_Orbiter>`__).
+Our `~sunpy.coordinates.frames.HeliographicCarrington` coordinate frame is now expressly intended for the co-alignment of images of the Sun's surface from different observatories.
+`~sunpy.coordinates.frames.HeliographicCarrington` now requires the specification of the observer location (Earth or otherwise) because the light travel time between the Sun and the observer is accounted for.
+SunPy output now matches the calculations by `JPL Horizons <https://ssd.jpl.nasa.gov/?horizons>`__ and `SPICE <https://naif.jpl.nasa.gov/naif/>`__.
+There may be small differences compared to Carrington coordinates computed by groups that do not use modern parameter values or the same assumptions for the methodology.
+
+Importantly, the Carrington longitude that is now calculated (including using :func:`sunpy.coordinates.sun.L0`) with not match earlier versions of SunPy.
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:64]:
```suggestion
Importantly, the Carrington longitude that is now calculated (including using :func:`sunpy.coordinates.sun.L0`) will not match earlier versions of SunPy.
```

File: docs/whatsnew/2.0.rst [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        (0., 0., 695700.)>,
+    <HeliographicStonyhurst Coordinate (obstime=None): (lon, lat, radius) in (deg, deg, km)
+        (0.00277778, 0.00277778, 695700.)>)
+
+Download behind proxies
+=======================
+
+With the release of parfive 1.1, sunpy has been patched now to able to pass proxy information to `aiohttp.ClientSession`.
+
+* Proxy URL is read from the environment variables `HTTP_PROXY` or `HTTPS_PROXY`, depending on the protocol of the `url` passed.
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:134]:
```suggestion
* Proxy URL is read from the environment variables `HTTP_PROXY` or `HTTPS_PROXY`.
```

File: docs/whatsnew/2.0.rst [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    $ HTTPS_PROXY=https://user:password@proxyserver.com:3128
+    $ export HTTP_PROXY
+    $ export HTTPS_PROXY
+
+these will be used to enable downloads through a proxy.
+
+Citation update
+===============
+
+We had a new paper accepted in, The Astrophysical Journal and you can find the bibtex for it by::
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:152]:
```suggestion
A paper discussing sunpy 1.0 was accepted in The Astrophysical Journal and you can find the bibtex for it by running::
```

File: docs/whatsnew/2.0.rst [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    >>> import sunpy
+    >>> sunpy.__citation__
+
+or `accessing the website directly <https://iopscience.iop.org/article/10.3847/1538-4357/ab4f7a>`__.
+
+aiaprep is now deprecated
+=========================
+
+With the release of the new `aiapy <https://aiapy.readthedocs.io>`__ package, `sunpy.instr.aia.aiaprep` will be removed in version 2.1.
+You can use the `register` function in the aiapy package to convert AIA images to level 1.5.
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:163]: @wtbarnes is the wording here correct? I would have thought you needed to run the other functions in aiaprep to get close to 1.5? Is something like "to align AIA images" better?

File: docs/whatsnew/2.0.rst [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+Search Attributes
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:165]: I propose we stick this at the top. It's probably our largest feature this release.

File: docs/whatsnew/2.0.rst [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+To search with `~sunpy.net.Fido`, you need to specify attributes to search against.
+Before sunpy 2.0, you had to supply values, as the following example demonstrates::
+
+    >>> from sunpy.net import Fido, attrs as a
+    >>> Fido.search(a.Time('2012/3/4', '2012/3/6'), a.Instrument("norh"), a.Wavelength(17*u.GHz))
+
+There was no way to know if the value was correct, but now we have a extenstive list of supported values from the clients and servers we can request data from.
+
+Using ``Instrument`` as an example, if you print the object::
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:176]:
```suggestion
Using `~sunpy.net.attrs.Instrument` as an example, if you print the object::
```

**Comment by ayshih** [comment]
My inclination is to delete the section about `get_rectangle_coordinates()`.  It gives too much visibility to a utility that I anticipate very few users using directly.

At the very least, the section should be redirected to discuss the unifying of the API of `submap()` and `draw_rectangle()`, with the utility being an afterthought at most.  That has much more direct impact on users.

**Comment by Cadair** [comment]
Yeah, I agree swapping that out for a section on `submap`/`draw_rectangle` achieves the same goal while being much more user facing.

**Comment by nabobalis** [comment]
Can someone draft a paragraph on ` submap/draw_rectangle`?

File: docs/whatsnew/2.0.rst [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+************************
+What's New in SunPy 2.0?
+************************
+
+Overview
+========
+
+The SunPy project is pleased to announce the 2.0 release of the sunpy package.
+
+.. _whatsnew-2.0-pixels:
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:14]: wait what's this doing here.

File: docs/whatsnew/2.0.rst [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+    >>> aia = sunpy.map.Map(AIA_171_IMAGE)
+
+    >>> bottom_left = SkyCoord(-100 * u.arcsec, -100 * u.arcsec, frame=aia.coordinate_frame)
+    >>> width = 500 * u.arcsec
+    >>> height = 300 * u.arcsec
+
+    >>> sub_aia = aia.submap(bottom_left, width=width, height=height)
+
+    >>> fig = plt.figure()
+    >>> ax1 =fig.add_subplot(1, 2, 1, projection=aia)
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:105]:
```suggestion
    >>> ax1 = fig.add_subplot(1, 2, 1, projection=aia)
```

File: docs/whatsnew/2.0.rst [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+SunPy output now matches the calculations by `JPL Horizons <https://ssd.jpl.nasa.gov/?horizons>`__ and `SPICE <https://naif.jpl.nasa.gov/naif/>`__.
+There may be small differences compared to Carrington coordinates computed by groups that do not use modern parameter values or the same assumptions for the methodology.
+
+Importantly, the Carrington longitude that is now calculated (including using :func:`sunpy.coordinates.sun.L0`) will not match earlier versions of SunPy.
+A detailed write-up of the calculation approach and comparisons to other resources can be found :ref:`here <sunpy-coordinates-carrington>`.
+
+Download behind proxies
+=======================
+
+With the release of parfive 1.1, sunpy has been patched now to able to pass proxy information to `aiohttp.ClientSession`.
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:164]:
```suggestion
With the release of parfive 1.1, sunpy has been patched to be able to utilize proxy servers when downloading files.
```

File: docs/whatsnew/2.0.rst [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+===============
+
+A paper discussing sunpy 1.0 was accepted in The Astrophysical Journal and you can find the bibtex for it by running::
+
+    >>> import sunpy
+    >>> sunpy.__citation__
+
+or `accessing the website directly <https://iopscience.iop.org/article/10.3847/1538-4357/ab4f7a>`__.
+
+aiaprep is now deprecated
```
Comment (Cadair) [inline docs/whatsnew/2.0.rst:191]: now I think this is too low down the page :see_no_evil:

Comment (wtbarnes) [reply docs/whatsnew/2.0.rst:191]: Yeah I'd agree. It is a small change, but something that people obviously will want to know about

File: docs/whatsnew/2.0.rst (hunk 6 above)
Comment (wtbarnes) [reply docs/whatsnew/2.0.rst:195]:
```suggestion
Equivalent functionality is provided by the `register` function in aiapy. For more details,
see the `example on registering and aligning level 1 AIA images <https://aiapy.readthedocs.io/en/latest/generated/gallery/prepping_level_1_data.html#registering-and-aligning-level-1-data>`_
in the aiapy documentation.
```

**Comment by dstansby** [comment]
I approve (although I can't approve it through the github UI as I'm technically the PR author)

**Pull request merged**